web/src/message_scroll.js

```diff
@@ -163,21 +163,23 @@ export function initialize() {
                 // Mark new selected message as read
                 messages = [event.msg_list.get(event.id)];
             }
-            if (event.msg_list.can_mark_messages_read()) {
-                unread_ops.notify_server_messages_read(messages, {from: "pointer"});
-            } else if (
-                unread.get_unread_messages(messages).length !== 0 &&
-                // The below checks might seem redundant, but it's
-                // possible this logic, which runs after a delay, lost
-                // a race with switching to another view, like Recent
-                // Topics, and we don't want to display this banner
-                // in such a view.
-                //
-                // This can likely be fixed more cleanly with another approach.
-                narrow_state.filter() !== undefined &&
-                message_lists.current === event.msg_list
-            ) {
-                unread_ui.notify_messages_remain_unread();
+            if (unread_ops.is_window_focused()) {
```

Comment:
Should this instead be making the conditional above be `if event.mark_read && unread_ops.is_window_focused()`, to avoid unnecessary extra nesting?

Reply:
I guess my other potential concern about this change is that if you do some action (like say hit `pagedown`) and then immediately hit alt-tab quickly to defocus the window before the scroll finishes processing, then I think we will no longer mark messages as read? And the new `process_selected_message` logic seems like it will only do the final selected message, not the N that you scrolled past.

I wonder whether we instead want logic of the form "store the set of messages that were scrolled past" in an else clause here, and then instead of the new `process_selected_message` call, mark-as-read the set of messages that had been stored for such adjustment when the window is focused again.

Not sure, I certainly would love to avoid that sort of action-at-a-distance complexity. An alternative approach would be to instead have the check here be something like "If the window has been focused within the last 60 seconds" rather than whether it's focused right now.